sklearn/model_selection/_search_successive_halving.py

```diff
@@ -0,0 +1,863 @@
+              spawning of the jobs
+
+            - An int, giving the exact number of total jobs that are
+              spawned
+
+            - A string, giving an expression as a function of n_jobs,
+              as in '2*n_jobs'
+
+    cv : int, cross-validation generator or an iterable, optional (default=5)
+        Determines the cross-validation splitting strategy.
+        Possible inputs for cv are:
+
+        - integer, to specify the number of folds in a `(Stratified)KFold`,
+        - :term:`CV splitter`,
+        - An iterable yielding (train, test) splits as arrays of indices.
+
+        For integer/None inputs, if the estimator is a classifier and ``y`` is
+        either binary or multiclass, :class:`StratifiedKFold` is used. In all
+        other cases, :class:`KFold` is used.
+
+        Refer :ref:`User Guide <cross_validation>` for the various
+        cross-validation strategies that can be used here.
+
+    refit : boolean, default=True
+        If True, refit an estimator using the best found parameters on the
+        whole dataset.
+
+        The refitted estimator is made available at the ``best_estimator_``
+        attribute and permits using ``predict`` directly on this
+        ``GridSearchCV`` instance.
+
+    verbose : integer
+        Controls the verbosity: the higher, the more messages.
+
+    error_score : 'raise' or numeric
+        Value to assign to the score if an error occurs in estimator fitting.
+        If set to 'raise', the error is raised. If a numeric value is given,
+        FitFailedWarning is raised. This parameter does not affect the refit
+        step, which will always raise the error. Default is ``np.nan``
+
+    return_train_score : boolean, default=False
+        If ``False``, the ``cv_results_`` attribute will not include training
+        scores.
+        Computing training scores is used to get insights on how different
+        parameter settings impact the overfitting/underfitting trade-off.
+        However computing the scores on the training set can be computationally
+        expensive and is not strictly required to select the parameters that
+        yield the best generalization performance.
+
+    max_budget : int, optional(default='auto')
+        The maximum number of resources that any candidate is allowed to use
+        for a given iteration. By default, this is set ``n_samples`` when
+        ``budget_on='n_samples'`` (default), else an error is raised.
+
+    budget_on : `n_samples` or str, optional(default='n_samples')
+        Defines the nature of the budget. By default, the budget is the number
+        of samples. It can also be set to any parameter of the base estimator
+        that accepts positive integer values, e.g. 'n_iterations' or
+        'n_estimators' for a gradient boosting estimator. In this case
+        ``max_budget`` cannot be 'auto'.
+
+    ratio : int or float, optional(default=3)
```

Comment:
perhaps `reduction` or `reduction_rate` or `reduce_by` or `divide_by` or `divisor`; perhaps `elimination_rate`...

Then again, I'm not sure how to express that this is both the rate of reduction and the number of target candidates.

I also think if we call this `*Halving*SearchCV` this must default to 2 or users will not realise.